app/modal-windows/components/dynamic-component/dynamic-component.component.ts

```diff
@@ -0,0 +1,53 @@
+// File: dynamic-component.component.ts
+// Created by: Daniel James
+// Date created: January 30, 2017
+//
+
+
+import { Component, ViewContainerRef, ViewChild, ComponentFactoryResolver, ComponentRef, OnInit, Type } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { WindowService } from './../../services/window.service';
+
+import { NewFileComponent } from './../../../components/new-file/new-file.component'; //TODO, shorter way to do this...?
+
+/**
+ * This component loads one of the components specified in componentMap depending on ModalInfo.kind for this window.
+ */
+@Component({
+    selector: 'sinap-dynamic-component',
+    entryComponents: [NewFileComponent],
+    template: `<div #container></div>`,
+    providers: [WindowService]
+})
+export class DynamicComponent {
+    private currentComponent: ComponentRef<any>;
+
+    @ViewChild('container', { read: ViewContainerRef })
+    private container: ViewContainerRef;
+
+    // TODO: I wish there was a better way to do this.
```

Comment:
Decorators will let you register the components and their types to this dictionary automatically. 

(See how I do this in core.ts)